mempool.go

```diff
@@ -1234,14 +1242,36 @@ func (mp *txMemPool) LastUpdated() time.Time {
 	return mp.lastUpdated
 }
 
+// janitorHandler scans the mempool at the interval specified by
+// janitorScanInterval removing any transactions that are older
+// than the duration specified by janitorMaxAge.
+func (mp *txMemPool) janitorHandler() {
+	for {
+		time.Sleep(janitorScanInterval)
+
+		mp.Lock()
+		for _, desc := range mp.pool {
+			if time.Since(desc.Added) > janitorMaxAge {
+				mp.removeTransaction(desc.Tx)
```

Comment:
I'm worried about this modifying mp.pool as it's being iterated over.  I believe that if a deletion cause a reallocation (to compact the hash table), we may miss iterating over some transactions.